packages/buidler-core/src/internal/core/providers/util.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  BuidlerNetworkAccount,
+  BuidlerNetworkHDAccountsConfig,
+} from "../../../types";
+import { deriveKeyFromMnemonicAndPath } from "../../util/keys-derivation";
+import { DEFAULT_BUIDLER_NETWORK_BALANCE } from "../config/default-config";
+import { BuidlerError } from "../errors";
+import { ERRORS } from "../errors-list";
+
+const HD_PATH_REGEX = /^m(:?\/\d+'?)+\/?$/;
+
+export function derivePrivateKeys(
```

Comment:
Shouldn't this function be used in `HDWalletProvider`'s constructor?

Reply:
Yes, I created it to reuse it there, and forgot about that.